src/Core/Tamper.hpp

```diff
@@ -42,6 +42,50 @@ namespace SokuLib
 		*reinterpret_cast<PBYTE>(addr + 0) = 0xE8;
 		TamperNearJmpOpr(addr, target);
 	}
+
+	// Trampoline - can hook function entries so all calls run the custom code
+	//   addr - address where to insert code
+	//   target - address of the custom code
+	//   offset - how many bytes to replace, used to align assembly OPs,
+	//          minimal required offset is 5(five)
+	//   return - allocated data that runs the original code.
+	DWORD TrampolineCreate(DWORD addr, DWORD target, int offset) {
```

Comment:
I would probably make this a template that takes as the type of the function pointer that is replaced and return it to prevent casting all the time.
Maybe something like this ?
```CPP
template<typename T>
T TrampolineCreate(T addr, unsigned char *target, int offset)
```

You could even be very fancy and make an object for this so you have less chance of messing up your call to Remove (or maybe keep those 2 and make the object as an helper) by overloading the destructor and operator().

In any case, if you prefer it like this, you should at least either make this inline or inside the .cpp because I'm pretty sure it will cause linker error if included in multiple CPP files.

Reply:
ah, it was supposed to be inline.
you may change it as you like, i don't have preferences.

Reply:
ok just make it inline for now and I'll probably make the object approach